src/Transform/LowerKrnlShape.cpp

```diff
@@ -42,21 +42,29 @@ class LowerKrnlShape : public OpRewritePattern<KrnlShapeOp> {
   LogicalResult matchAndRewrite(
       KrnlShapeOp krnlShapeOp, PatternRewriter &rewriter) const override {
     auto loc = krnlShapeOp.getLoc();
-    auto rank =
+    int64_t rank =
         krnlShapeOp.alloc().getType().dyn_cast<MemRefType>().getShape().size();
 
+    // Create MemRef to hold shape information.
+    auto memRefType = MemRefType::get({rank}, rewriter.getIndexType());
+    auto newMemRefAlloc = rewriter.create<AllocOp>(loc, memRefType);
+
     SmallVector<mlir::Value, 4> fromExtentsOpOperands;
     for (int idx = 0; idx < rank; idx++) {
       auto index = rewriter.create<ConstantOp>(
           loc, rewriter.getIntegerAttr(rewriter.getIndexType(), idx));
       auto operand = rewriter.create<KrnlDimOp>(
           loc, rewriter.getIndexType(), krnlShapeOp.alloc(), index);
       fromExtentsOpOperands.emplace_back(operand);
+
+      // Store value in the new MemRef.
+      auto idxValue =
```

Comment:
nit: no need to use auto when "Value" is just as short. Let's use them only for large types